test/errorshow.jl

```diff
@@ -808,6 +808,64 @@ end
 @test_throws ArgumentError("invalid index: \"foo\" of type String") [1]["foo"]
 @test_throws ArgumentError("invalid index: nothing of type Nothing") [1][nothing]
 
+# issue #53618
+@testset "FieldErrorHint" begin
+    struct FieldFoo
+        a::Float32
+        b::Int64
+    end
+
+    s = FieldFoo(1, 2)
+
+    @test s.a == 1.0f0
+    @test s.b == 2
+    @test_throws FieldError s.c
+
+    ex = try
+        s.c
+    catch e
+        e
+    end::FieldError
```

Comment:
```suggestion
    test = @test_throws FieldError s.c

    ex = test.value::FieldError
```
`@test_throws` saves the exception